comp/dogstatsd/server/parse.go

```diff
@@ -206,9 +201,17 @@ func (p *parser) parseMetricSample(message []byte) (dogstatsdMetricSample, error
 				return dogstatsdMetricSample{}, fmt.Errorf("dogstatsd timestamp should be > 0")
 			}
 			timestamp = time.Unix(ts, 0)
-		// container ID
+		// local data
 		case p.dsdOriginEnabled && bytes.HasPrefix(optionalField, localDataPrefix):
-			containerID = p.resolveContainerIDFromLocalData(optionalField)
+			rawLocalData := string(optionalField[len(localDataPrefix):])
+			localData, err = origindetection.ParseLocalData(rawLocalData)
+			if err != nil {
+				return dogstatsdMetricSample{}, fmt.Errorf("failed to parse OriginInfo.LocalData %s: %v", rawLocalData, err)
```

Comment:
An type name, being an implementation detail, in the error message is not very informative for the users. Would it be possible to rephrase this in terms that are present in the [protocol description](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics)?

Reply:
Sure! I've rephrased it here: https://github.com/DataDog/datadog-agent/pull/32757/commits/85a48fe6ddbe3e3f1d265a21b7d353d6bc71045a